<launch>

  <node pkg="occgrid_planner" type="occgrid_planner" name="planner" output="screen">
    <param name="neighbourhood" value="8"/>
    <param name="base_frame" value="/body"/>
    <param name="debug" value="False"/>
    <remap from="/planner/occ_grid" to="/map"/>
    <remap from="/planner/goal" to="/move_base_simple/goal"/>
  </node>


  <node pkg="occgrid_planner" type="path_optimizer" name="path_optimizer" output="screen">
      <param name="max_acceleration" value="0.3"/> 
      <param name="max_braking" value="0.1"/> 
      <param name="velocity" value="0.1"/> 
      <remap from="/path_optimizer/path" to="/planner/path"/>
  </node>

  <node pkg="occgrid_planner" type="path_follower" name="path_follower" output="screen">
      <param name="Kx" value="1.0"/>
      <param name="Ky" value="1.0"/>
      <param name="Ktheta" value="1.0"/>
      <param name="max_rot_speed" value="1.0"/>
      <param name="max_velocity" value="1.0"/>
      <param name="max_y_error" value="1.0"/>
      <param name="look_ahead" value="1.0"/>
      <param name="base_frame" value="/body"/>
      <remap from="/path_follower/traj" to="/path_optimizer/trajectory"/>
      <remap from="/path_follower/twistCommand" to="/mux/autoCommand"/>
  </node>

  <node pkg="topic_tools" type="mux" name="cmd_mux" 
      args="/vrep/twistCommand /teleop/twistCommand /mux/autoCommand">
  </node>

  <node pkg="vrep_ros_teleop" type="teleop_mux.py" name="teleop_mux">
      <param name="joystick_button" value="0" type="int"/>
      <param name="joystick_topic" value="/teleop/twistCommand"/>
      <param name="auto_button" value="1" type="int"/>
      <param name="auto_topic" value="/mux/autoCommand"/>
      <remap from="/teleop_mux/joy" to="/joy"/>
  </node>
  <!-- Axes -->
  <node pkg="vrep_ros_teleop" type="teleop.py" name="teleop">
    <param name="axis_linear_x" value="1" type="int"/>
    <param name="axis_angular" value="0" type="int"/>
    <param name="scale_linear" value="0.5" type="double"/>
    <param name="scale_angular" value="1.0" type="double"/>
    <remap from="/teleop/joy" to="/joy"/>
  </node>


</launch>
